orion-proxy/src/proxy.rs

```diff
@@ -291,8 +291,12 @@ async fn spawn_services(info: ServiceInfo) -> Result<()> {
     let secret_manager_clone = secret_manager.clone();
     set.spawn(async move {
         configure_initial_resources(bootstrap_clone, listener_factories, configuration_senders_clone.clone()).await?;
-        let xds_handler = XdsConfigurationHandler::new(secret_manager_clone, configuration_senders_clone);
-        _ = xds_handler.xds_run(node, clusters, ads_cluster_names).await;
+        let mut xds_handler = XdsConfigurationHandler::new(secret_manager_clone, configuration_senders_clone);
+        let res = xds_handler.xds_run_loop(node, clusters, ads_cluster_names).await;
```

Comment:
we should not start the loop if ads_cluster_names unset

Reply:
Correct, we keep skip it.